homeassistant/components/esphome/config_flow.py

```diff
@@ -150,33 +151,45 @@ def _name(self, value: str) -> None:
         self.context["title_placeholders"] = {"name": self._name}
 
     async def _async_try_fetch_device_info(self) -> FlowResult:
-        error = await self.fetch_device_info()
-
-        if error == ERROR_REQUIRES_ENCRYPTION_KEY:
+        """Try to fetch device info and return any errors."""
+        response: str | None
+        if self._noise_required:
+            # If we already know we need encryption, don't try to fetch device info
+            # without encryption.
+            response = ERROR_REQUIRES_ENCRYPTION_KEY
+        else:
+            # After 2024.08, stop trying to fetch device info without encryption
+            # so we can avoid probe requests to check for password. At this point
+            # most devices should announce encryption support and password is
+            # deprecated and can be discovered by trying to connect only after they
+            # interact with the flow since it is expected to be a rare case.
```

Comment:
We still have to probe for passwords but since its deprecated, in a year we can avoids the probes completely when we have zeroconf properties and only probe for password when they interact with the config flow